tests/vm_test.go

```diff
@@ -1969,12 +1969,25 @@ status:
 
 			By("Blocking virt-handler from reconciling the VMI")
 			libpod.AddKubernetesApiBlackhole(getHandlerNodePod(virtClient, nodeName), componentName)
-			Eventually(getHandlerNodePod(virtClient, nodeName).Items[0], 120*time.Second, time.Second, HaveConditionFalse(k8sv1.PodReady))
+			Eventually(func() k8sv1.Pod {
+				return getHandlerNodePod(virtClient, nodeName).Items[0]
+			}, 120*time.Second, time.Second).Should(HaveConditionFalse(k8sv1.PodReady))
 
-			DeferCleanup(func() {
+			defer func() {
 				libpod.DeleteKubernetesApiBlackhole(getHandlerNodePod(virtClient, nodeName), componentName)
```

Comment:
These kind of hacks are causing the whole tests suite to collapse on itself.

These is nothing safe that can be done to fix something like this and I hope to see this whole test removed.

But this is not really related to this fix. I just think the fix is to remove it.

Reply:
Dropped the usage of  `defer` for this purpose. I agree, it's not elegant. Thanks for the input.

Reply:
I was actually referring to hacking the virt-handler network.

It is indeed nicer without the `defer`, but what actually is done here is very risky and IMO does not fit our tests.
If for some reason the pod state is not fixed (e.g. the route is not deleted), we are left with a broken cluster and following tests will flake in a hard to understand manner.

IMO we should not have such tests around, we cannot cover every scenario in a reasonable safe manner.
Our resources are limited, tests are long and flaky enough not to risk them further.
If someone thinks it is worth the resources and risk, it would be better to allocate a dedicate job for such tests.  I think reality dictates that we need to manage our e2e tests and focus on the main functionality and leave the edge scenarios to the community to detect (reporting bugs, running more tests on D/S projects and contributing back).

Reply:
There is precedence for destructive tests, if we want to run them on separate lanes or not have them at all, the decision should be made in wider consensus + maintainers.

I am not sure what is better about the `AfterEach`, other than it is wrong afaik.

Note: I don't agree. This is not a conformance test and if anything this does fail loudly!

Reply:
Well, I disagreeing with the presence of cluster-level destructive tests and I reasoned about it in the prev message. The fact that they exist is not a sign of them being good for the overall project.

> This is not a conformance test and if anything this does fail loudly!

If tests like this fail, they either mess up all following tests or destabilize the cluster in ways we may not predict.

Reply:
> There is precedence for destructive tests, if we want to run them on separate lanes or not have them at all, the decision should be made in wider consensus + maintainers.

> Well, I disagree with the presence of cluster-level destructive tests and I reasoned about it in the prev message. The fact that they exist is not a sign of them being good for the overall project.

> > This is not a conformance test and if anything this does fail loudly!

> If tests like this fail, they either mess up all following tests or destabilize the cluster in ways we may not predict.


We have no rush to get this PR merged and from my point of view these observations are worth to be discussed like @xpivarc proposed. What would be a good way of shifting this discussion away from the PR to a wider forum?

Reply:
I will be drafting a proposal for the wider kubevirt community to move these kind of `destructive` and edge case tests to their own separate test lane and out of the gating lanes. We are currently testing everything in the gating lanes including severe edge cases which does not make sense and makes CI very costly. 

We will look at how to separate out these tests and where we want to run them.